### PR TITLE
[Bugfix:InstructorUI] Fix gradeable deletion

### PR DESCRIPTION
**Pull request by lavalleeale**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

Deleting a gradeable can possibly throw a frog robot due to peer grading not being deleted.

### What is the new behavior?
Deleting a gradeable will work successfully.

### Other information?

Testing requires editing the [canDelete](https://github.com/Submitty/Submitty/blob/3b9c945261a67cbc1a2b864a1404d0f56d48b2fa/site/app/models/gradeable/Gradeable.php#L1804) function to allow the deletion button to be shown in the UI.

### CODE REVIEW

**Review by williamschen23 — APPROVED** [review]
I tested the change, so I deleted a gradeable and looking at the sql, it deleted the gradeable_id from the gradeable table, the gradeable in the electronic_gradeable table, and the anon_id in the gradeable_anon table

File: migration/migrator/migrations/course/20241108162239_cascade_on_delete_gradeable.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE gradeable_anon DROP CONSTRAINT gradeable_anon_g_id_fkey")
+    database.execute("ALTER TABLE gradeable_anon ADD CONSTRAINT gradeable_anon_g_id_fkey FOREIGN KEY (g_id) REFERENCES gradeable(g_id) ON UPDATE CASCADE ON DELETE CASCADE")
+
+    database.execute("ALTER TABLE peer_grading_panel DROP CONSTRAINT peer_grading_panel_g_id_fkey")
+    database.execute("ALTER TABLE peer_grading_panel ADD CONSTRAINT peer_grading_panel_g_id_fkey FOREIGN KEY (g_id) REFERENCES public.electronic_gradeable(g_id) ON UPDATE CASCADE ON DELETE CASCADE")
+    pass
```
Comment (ziesski) [inline migration/migrator/migrations/course/20241108162239_cascade_on_delete_gradeable.py:22]:
```suggestion
        database.execute("""
            ALTER TABLE gradeable_anon DROP CONSTRAINT gradeable_anon_g_id_fkey;
            ALTER TABLE gradeable_anon ADD CONSTRAINT gradeable_anon_g_id_fkey
                FOREIGN KEY (g_id) REFERENCES gradeable(g_id)
                ON UPDATE CASCADE ON DELETE CASCADE;

            ALTER TABLE peer_grading_panel DROP CONSTRAINT peer_grading_panel_g_id_fkey;
            ALTER TABLE peer_grading_panel ADD CONSTRAINT peer_grading_panel_g_id_fkey
                FOREIGN KEY (g_id) REFERENCES public.electronic_gradeable(g_id)
                ON UPDATE CASCADE ON DELETE CASCADE;
        """)
```

Comment (ziesski) [reply migration/migrator/migrations/course/20241108162239_cascade_on_delete_gradeable.py:22]: We should wrap migration sql commands in one transaction.

**Review by ziesski — CHANGES_REQUESTED** [review]

**Review by ziesski — APPROVED** [review]
ON DELETE CASCADE is reasonable approach. Code looks good.

**Pull request merged**